compiler/main/checks.cpp

```diff
@@ -622,6 +624,16 @@ checkResolveRemovedPrims(void) {
   }
 }
 
+static void checkNoRecordDeletes() {
+  // No need to do for_alive_in_Vec - there shouldn't be any, period.
```

Comment:
for_alive_in_Vec does forv_Vec, except it does not execute the loop body for those nodes that have been "removed" from the AST. Such nodes are typically not pertinent. Here I am saying that there should not be record deletes even on non-pertinent nodes. Makes sense?